src/Components/EventCreation/Form/Form.js

```diff
@@ -9,42 +9,93 @@ import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
-import { createEvent } from "../../../queries";
+import { createEventMutation } from "../../../queries";
 
 const Form = ({logoutUser, loggedInUser}) => {
   const [game, setGame] = useState('');
   const [category, setCategory] = useState('');
-  const [address, setAddress] = useState();
-  const [city, setCity] = useState();
-  const [state, setState] = useState();
-  const [zip, setZip] = useState();
-  const [date, setDate] = useState();
-  const [startTime, setStartTime] = useState();
-  const [endTime, setEndTime] = useState();
-  const [eventDescription, setEventDescription] = useState();
-
-  const handleSubmit = (event) => {
-    event.preventDefault();
-
-    createNewEvent({
-      variables: {
-        input: {
-          date: date,
-          address: address,
-          city: city,
-          state: state,
-          zip: zip,
-          title: game,
-          description: eventDescription,
-          hostId: loggedInUser,
-        },
-      },
-    });
+  const [address, setAddress] = useState('');
+  const [city, setCity] = useState('');
+  const [state, setEState] = useState('');
+  const [zip, setZip] = useState(null);
+  const [date, setDate] = useState(null);
+  const [startTime, setStartTime] = useState(null);
+  const [endTime, setEndTime] = useState(null);
+  const [eventDescription, setEventDescription] = useState('');
+
+  const [createEvent, { data, loading, error }] = useMutation(createEventMutation);
+
+  const onCreateEvent = async(input) => {
+    try {
+      const { data } = await createEvent({ variables: { input } });
+      console.log(data);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+
+  const handleGameChange = (e) => {
+    setGame(e.target.value);
+  };
+  
+  const handleDateChange = (date) => {
+    setDate(date);
+  };
+
+  const handleStartTimeChange = (time) => {
+    setStartTime(time);
+  };
+
+  const handleEndTimeChange = (time) => {
+    setEndTime(time);
+  };
+
+  const handleCategoryChange = (e) => {
+    setCategory(e.target.value);
+  };
+
+  const handleAddressChange = (e) => {
+    setAddress(e.target.value);
+  };
+
+  const handleCityChange = (e) => {
+    setCity(e.target.value);
+  };
+
+  const handleStateChange = (e) => {
+    setEState(e.target.value);
+  };
+
+  const handleZipChange = (e) => {
+    setZip(e.target.value);
+  };
+
+  const handleEventDescriptionChange = (e) => {
+    setEventDescription(e.target.value);
+  };
+
+  const handleSubmit = () => {
+    const newEvent = {
+      date: dayjs(date).format("YYYY/MM/DD"),
+      address: address,
+      city: city,
+      state: state,
+      zip: parseInt(zip),
+      title: game,
+      description: eventDescription,
+      host: loggedInUser,
+      game: 1,
+      gameType: category,
+      startTime: dayjs(startTime).format("HH:mm:ss"),
+      endTime: dayjs(endTime).format("HH:mm:ss"),
+    }    
+    console.log(newEvent)
+    onCreateEvent(newEvent);
   };
```

Comment:
You did an excellent job creating these handle functions, which have enabled the mutation to function correctly.